.github/workflows/ci.yml

```diff
@@ -8,12 +8,28 @@ on:
 
 jobs:
   build:
-    runs-on: windows-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@main
       - name: Setup .NET
         uses: actions/setup-dotnet@main
         with:
           global-json-file: ./global.json
-      - name: dotnet test
-        run: dotnet test ./src/nunit.analyzers.tests/
+
+      - name: .NET6.0 Test
+        run: |
+            dotnet tool install --global dotnet-coverage
+            dotnet test --collect:"XPlat Code Coverage" -f net6.0 ./src/nunit.analyzers.tests/
+            dotnet-coverage merge -o merged.cobertura.xml -f cobertura -r coverage.cobertura.xml
+
+      - name: .NET Framework Test
+        if: runner.os == 'Windows'
+        run: |
+            dotnet test -f net462 ./src/nunit.analyzers.tests/
+
+      - name: Post Coverage
+        uses: k1LoW/octocov-action@v0
+        if: runner.os == 'Linux'
```

Comment:
This needs align with the https://github.com/nunit/nunit/pull/4286/ to use coverlet.